14-microbiome-diversity.Rmd

```diff
@@ -132,9 +150,19 @@ Sample similarities can be visualized on a lower-dimensional display
 additional information using variations in colour, shape or size.
 
 ```{r plot-mds-bray-curtis, fig.cap="MDS plot based on the Bray-Curtis distances on the GlobalPattern dataset."}
-plotReducedDim(se, "MDS_BC", colour_by = "SampleType")
+# Create ggplot object
+p <- plotReducedDim(se, "MDS_BC", colour_by = "SampleType")
+
+# Add explained variance for each axis
+e <- attr(reducedDim(se, "MDS_BC"), "eig");
+rel_eig <- e/sum(e[e>0])		  
```

Comment:
This calculates explained variance, and this is only defined for PCA and MDS (aka PCoA) and not available for most other ordination methods.

Now, the problem is that this needs to be done manually and in a different way for PCA and MDS. It would be easier if the explained variance could be made more readily available for visualization and analysis purposes, for instance through a generic method that operates on the se object. This should be available for both PCA and MDS at least (possibly also for other covariance based methods that have this defined).

The plotting function could then automatically detect the availability of this information and to the plot as it is a recommended practice to show the explained variance when it is available. It could be even automatically (and optionally) used to tune the ratio between X and Y axis widths, which has been sometimes recommended.

Reply:
`runPCA`/`calculatePCA` calculates the variance explained automatically. It is also included, if you plot `plotPCA`.

For `runMDS` it is different